src/flb_sds.c

```diff
@@ -407,3 +407,31 @@ void flb_sds_destroy(flb_sds_t s)
     head = FLB_SDS_HEADER(s);
     flb_free(head);
 }
+
```

Comment:
I think this function is not necessary and we can simply rely on flb_sds_printf() functionality. 

flb_sds_t by default always reallocate if necessary, and calculating the final length can be done with flb_sds_len()

Reply:
Functionality is a little different.

`flb_sds_printf` appends offset `flb_sds_len(s)` and it means appends string.
https://github.com/fluent/fluent-bit/blob/v1.8.10/src/flb_sds.c#L367

I want to overwrite flb_sds_t.
So if we use the API to overwrite, we need to call `flb_sds_len_set(sds, 0)` before the API.
```c
flb_sds_len_set(j_index, 0);
out_buf = flb_sds_printf(&j_index,
                        ES_BULK_INDEX_FMT_ID_WITHOUT_TYPE,
                        es_index,  id_key_str);
```

Reply:
ah I understand. thanks for the clarification.

On that case so I would suggest renaming the new function from:

- flb_sds_snprintf_realloc

to 

- flb_sds_snprintf

basically the "realloc" context is something that works behind the scenes, where the user don't need to be aware of it (similar to other sds...)

Reply:
Thank you for comment.
I renamed.